Resources/ProtobufDumper/ProtobufDumper/ProtobufCollector.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using google.protobuf;
+using ProtoBuf;
+
+namespace ProtobufDumper
+{
+    class ProtobufCollector
+    {
+        public List<FileDescriptorProto> Candidates { get; private set; }
+
+        public ProtobufCollector()
+        {
+            Candidates = new List<FileDescriptorProto>();
+        }
+
+
+        public bool CollectCandidate( string name, byte[] data )
```

Comment:
I can't tell what the return result actually means by looking at either the signature alone, or the function.

Reply:
This has been renamed to TryParseCandidate, returning detailed error data.